### PR TITLE
fix: Updates for MongoDB upgrade to V5

### PR DESCRIPTION
**Pull request by jnikhila**
* Updates to Custom MongoDB and Redis page
* Adds a Troubleshooting guide for Illegal instruction (core dumped)
* Moved Upgrade to Checkpoint section to Update Appsmith doc.
    * Formatting, readability and code block fixes

### CODE REVIEW

File: website/docs/getting-started/setup/instance-configuration/custom-mongodb-redis.md [hunk 1]
```diff
@@ -1,38 +1,83 @@
...
+Appsmith runs embedded instances of MongoDB and Redis. Appsmith uses embedded MongoDB to store application data and Redis to manage sessions. You can switch to external instances by changing the configuration. This page provides steps to set up Appsmith with external MongoDB and Redis. With external MongoDB and Redis, you can manage your application's scalability and performance.
 
-:::info Important
-If you are currently using an older version of MongoDB, it's important to upgrade to version 5.0 before connecting Appsmith to an external MongoDB instance. For more information, follow the steps detailed on MongoDB official documentation to [Upgrade a Replica Set to 5.0](https://www.mongodb.com/docs/manual/release-notes/5.0-upgrade-replica-set/).
+:::caution Attention
+To use a custom MongoDB with Appsmith v1.9.0 and later, you need MongoDB version 5 or higher. Follow the steps detailed in the MongoDB official documentation to [Upgrade a Replica Set to 5.0](https://www.mongodb.com/docs/manual/release-notes/5.0-upgrade-replica-set/).
```
Comment (sum35h) [inline website/docs/getting-started/setup/instance-configuration/custom-mongodb-redis.md:10]:
```suggestion
To use a custom MongoDB with Appsmith v1.9.0 and later, you need MongoDB version 5.x . Follow the steps detailed in the MongoDB official documentation to [Upgrade a Replica Set to 5.0](https://www.mongodb.com/docs/manual/release-notes/5.0-upgrade-replica-set/).
```

File: website/docs/getting-started/setup/instance-configuration/custom-mongodb-redis.md [hunk 2]
```diff
@@ -1,38 +1,83 @@
 ---
 sidebar_position: 3
+toc_max_heading_level: 4
 ---
-# Custom MongoDB & Redis
+# External MongoDB and Redis
 
-Appsmith runs it's own instances of MongoDB database for storing all application data, and Redis for storing session information. It's easy to switch to externally running instances of these services, by just changing the relevant environment variables.
+Appsmith runs embedded instances of MongoDB and Redis. Appsmith uses embedded MongoDB to store application data and Redis to manage sessions. You can switch to external instances by changing the configuration. This page provides steps to set up Appsmith with external MongoDB and Redis. With external MongoDB and Redis, you can manage your application's scalability and performance.
```
Comment (sum35h) [inline website/docs/getting-started/setup/instance-configuration/custom-mongodb-redis.md:7]:
```suggestion
Appsmith runs embedded instances of MongoDB and Redis. Appsmith uses embedded MongoDB to store application data and Redis to manage sessions. You can switch to external instances by changing the configuration. This page provides steps to set up Appsmith with external MongoDB and Redis. With external MongoDB and Redis, the Appsmith container becomes stateless, allowing better management of the instance in terms of performance and scalability.
```

File: website/docs/getting-started/setup/instance-management/update-appsmith.mdx [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+   #    - WATCHTOWER_LIFECYCLE_HOOKS=true
+   ```
+
+3. Save the file and run:
+
+   ```bash
+   docker-compose up -d
+   ```
+
+When the server restarts, Appsmith updates to the latest version. This setup runs an Appsmith and a Watchtower container. The Watchtower checks and updates Appsmith when a new version becomes available.
```
Comment (sum35h) [inline website/docs/getting-started/setup/instance-management/update-appsmith.mdx:87]:
```suggestion
This setup runs an Appsmith and a Watchtower container. The Watchtower checks and updates Appsmith when a new version becomes available. 
```

File: website/docs/help-and-support/troubleshooting-guide/deployment-errors.mdx [hunk 4]
```diff
@@ -135,6 +135,55 @@ As the error message in the server logs suggests, your Appsmith instance has mis
...
+
+<Message
+messageContainerClassName="error" 
+messageContent="Illegal instruction (core dumped) mongod –fork –port 27017 –dbpath “$MONGO_DB_PATH"></Message>
+
+#### Cause
+MongoDB version 5 has specific CPU requirements. If these requirements are not met the server wouldn't start.
+
+#### Solution
+To resolve this error, you must upgrade your infrastructure. See the [List of complaint hardware](https://www.mongodb.com/docs/manual/administration/production-notes/#platform-support).
```
Comment (sum35h) [inline website/docs/help-and-support/troubleshooting-guide/deployment-errors.mdx:155]: Currently, we have 2 options, either upgrade hardware to support Mongov5 or rollback to old Appsmith v.18.
Lets add a third option, that is to provision and use an external Mongov5 (preferably on Mongo Atlas)

File: website/docs/help-and-support/troubleshooting-guide/deployment-errors.mdx [hunk 5]
```diff
@@ -135,6 +135,55 @@ As the error message in the server logs suggests, your Appsmith instance has mis
...
+
+If you can't upgrade your infrastructure, downgrade your Appsmith instance to v1.8.15. Follow the steps below to downgrade to v1.8.15:
```
Comment (sum35h) [inline website/docs/help-and-support/troubleshooting-guide/deployment-errors.mdx:157]:
```suggestion
If you can't upgrade your infrastructure, we recommend using an externally hosted Mongo v5.x instance (preferably on [Mongo Atlas](https://cloud.mongodb.com).
If the above options are not feasible for you then you will have to downgrade your Appsmith instance to v1.8.15. Follow the steps below to downgrade to v1.8.15:
```

File: website/docs/getting-started/setup/instance-configuration/custom-mongodb-redis.md (hunk 1 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/instance-configuration/custom-mongodb-redis.md:10]: @sharat87 - Please share your thoughts here.

Comment (sharat87) [reply website/docs/getting-started/setup/instance-configuration/custom-mongodb-redis.md:10]: No, v6 is also okay.

File: website/docs/getting-started/setup/instance-configuration/custom-mongodb-redis.md [hunk 6]
```diff
@@ -1,38 +1,83 @@
...
 :::
 
-## Custom MongoDB
+## External MongoDB
+Follow the below steps to set up an external MongoDB:
 
-To point to a custom MongoDB, update the `env` variable called `APPSMITH_MONGODB_URI` to point to your custom MongoDB server. For example, if you are using [MongoDB Cloud](https://www.mongodb.com/cloud), you can change this to something like the following:
+### Prerequisites
+* Ensure that your external MongoDB has a replica set configuration in place. Connect to your database as an admin user and run [rs.initiate()](https://docs.mongodb.com/manual/reference/method/rs.initiate/). Note that if you're using MongoDB Cloud, the replica set configuration is already set up for you.
+* Ensure that the user you are connecting with has `readWrite` and `clusterMonitor` permissions.
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/custom-mongodb-redis.md:18]: It looks a bit confusing.  How about? 

```suggestion
* Ensure the MongoDB user account has `readWrite` and `clusterMonitor` roles assigned.

File: website/docs/help-and-support/troubleshooting-guide/deployment-errors.mdx (hunk 4 above)
Comment (danciaclara) [reply website/docs/help-and-support/troubleshooting-guide/deployment-errors.mdx:155]: @jnikhila can we add this?

**Review by danciaclara — CHANGES_REQUESTED** [review]

File: website/docs/getting-started/setup/instance-management/README.mdx [hunk 7]
```diff
@@ -1,115 +1,15 @@
...
-
- ```
- docker-compose up -d
-  ```
-:::note 
-If you haven't been using your Appsmith instance for a while, it could be possible that there might be multiple versions that show up in the error message. In such a case, perform the same steps for each version in order, starting from the oldest.
-:::
-
-
 ## Switching off auto updates
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-management/README.mdx:10]: Isn't this also related to updates? Until we figure out how Instance Management should look, can we not make any changes there? Else we will have make this a proper landing page. I think this should be handled in a different PR. as in the deploy preview this page looks and incomplete.

**Review by danciaclara — CHANGES_REQUESTED** [review]

File: website/docs/getting-started/setup/instance-configuration/custom-mongodb-redis.md [hunk 8]
```diff
@@ -1,38 +1,83 @@
 ---
 sidebar_position: 3
+toc_max_heading_level: 4
 ---
-# Custom MongoDB & Redis
+# External MongoDB and Redis
 
-Appsmith runs it's own instances of MongoDB database for storing all application data, and Redis for storing session information. It's easy to switch to externally running instances of these services, by just changing the relevant environment variables.
+Appsmith runs embedded instances of MongoDB and Redis. Appsmith uses MongoDB and Redis for data storage and session management. You can use external instances by configuring the instance. External instances are stateless and help in better performance and scalability. Follow the steps on this page to set up external MongoDB and Redis for Appsmith.
```
[the remark this answers is not in the document]
Comment (jnikhila) [reply website/docs/getting-started/setup/instance-configuration/custom-mongodb-redis.md:7]: > Let's combine some of these lines. Hope you didn't have to change it because of the readability checker? .Also using the word "embedded" here might clash with the separate Embedded DB we are going to have for self-hosted instances which might cause confusion in the future. 
> 
> 
> 
> ```suggestion
> 
> Appsmith runs its own instances of MongoDB for storing the application data and Redis for storing session information. You can also switch to external instances, which makes the Appsmith container stateless, enabling better performance and scalability. Follow the steps on this page to set up external MongoDB and Redis for Appsmith.
> 
> ```
> 
> 

The word embedded is suggested by Sumesh. I will check the content that you suggested.

File: website/docs/getting-started/setup/instance-management/README.mdx (hunk 7 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/instance-management/README.mdx:10]: That's a configuration task to turn on or off auto updates so haven't moved it.

File: website/docs/getting-started/setup/instance-configuration/custom-mongodb-redis.md [hunk 9]
```diff
@@ -1,38 +1,83 @@
...
 
-:::info Important
-If you are currently using an older version of MongoDB, it's important to upgrade to version 5.0 before connecting Appsmith to an external MongoDB instance. For more information, follow the steps detailed on MongoDB official documentation to [Upgrade a Replica Set to 5.0](https://www.mongodb.com/docs/manual/release-notes/5.0-upgrade-replica-set/).
+:::caution Attention
+To use a custom MongoDB with Appsmith v1.9.0 and later, you need MongoDB version 5 or higher. Follow the steps detailed in the MongoDB official documentation to [Upgrade a Replica Set to 5.0](https://www.mongodb.com/docs/manual/release-notes/5.0-upgrade-replica-set/).
```
[the remark this answers is not in the document]
Comment (jnikhila) [reply website/docs/getting-started/setup/instance-configuration/custom-mongodb-redis.md:10]: While both "on later" and "onwards" are used to indicate continuing from a certain point, "onwards" is more commonly used in British English. Since we are adhering to the rules and conventions of American English, we should use "on later" instead of "onwards." Also, technically `later` or `higher` terms signify that the compatibility is with v5 and versions that are released after v5. Let's discuss this if needed.

File: website/docs/getting-started/setup/instance-configuration/custom-mongodb-redis.md (hunk 6 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/instance-configuration/custom-mongodb-redis.md:18]: Rewriting this as - Ensure that the user you are connecting with has `readWrite` and `clusterMonitor` roles assigned. ( as we are in MongoDB section, so IMO we shouldn't add MongoDB to specify that)

File: website/docs/getting-started/setup/instance-management/README.mdx (hunk 7 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/instance-management/README.mdx:10]: We can work on the landing page in the incremental PR. But as the checkpoint upgrades were related, moving that page to the appropriate location made more sense. I can add some cards for now to the page so that it doesn't look odd.

File: website/docs/getting-started/setup/instance-configuration/custom-mongodb-redis.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (jnikhila) [reply website/docs/getting-started/setup/instance-configuration/custom-mongodb-redis.md:10]: done.

File: website/docs/help-and-support/troubleshooting-guide/deployment-errors.mdx [hunk 10]
```diff
@@ -135,6 +135,59 @@ As the error message in the server logs suggests, your Appsmith instance has mis
...
+messageContainerClassName="error" 
+messageContent="Illegal instruction (core dumped) mongod –fork –port 27017 –dbpath “$MONGO_DB_PATH"></Message>
+
+#### Cause
+MongoDB version 5 has specific CPU requirements. If these requirements are not met the server wouldn't start.
+
+#### Solution
+To resolve this error, you must upgrade your infrastructure. See the [List of complaint hardware](https://www.mongodb.com/docs/manual/administration/production-notes/#platform-support).
+
+If you can't upgrade your infrastructure or can't upgrade your embedded Appsmith MongoDB version:
```
Comment (sum35h) [inline website/docs/help-and-support/troubleshooting-guide/deployment-errors.mdx:157]:
```suggestion
If is not possible for you to upgrade your infrastructure to support MongoDB v5.
```

File: website/docs/help-and-support/troubleshooting-guide/deployment-errors.mdx [hunk 11]
```diff
@@ -135,6 +135,59 @@ As the error message in the server logs suggests, your Appsmith instance has mis
...
+    </TabItem>
+  </Tabs>
+
+
+  2. Run the restart command: 
+
+  ```bash
+  docker-compose up -d
+  ```
+Your server starts with version 1.8.15, which makes all the necessary schema changes to your Appsmith instance.
```
Comment (sum35h) [inline website/docs/help-and-support/troubleshooting-guide/deployment-errors.mdx:193]: Should we also call out that v1.8.15 misses out on all the latest features, and should only be seen as a temporary solution until the usersare able to provision a VM that supports Mongo v5 or provision an external MongoDB?

Comment (sum35h) [reply website/docs/help-and-support/troubleshooting-guide/deployment-errors.mdx:193]: v1.8.15 looks like a very bad path on the long term and should only be used to unblock user on the short term.
